code/acting/src/acting/velocity_publisher_dummy.py

```diff
@@ -39,20 +43,27 @@ def run(self):
         if not self.enabled:
             return
         self.loginfo('VelocityPublisherDummy node running')
+        start_time = rospy.get_time()
+        self.loginfo(f"Start Time: {start_time}")
 
         def loop(timer_event=None):
             """
             Publishes dummy velocity
             :param timer_event: Timer event from ROS
             :return:
             """
-            # self.loginfo('Published dummy velocity: ' + str(self.velocity))
-            self.velocity_pub.publish(self.velocity)
-            if self.velocity > self.max_velocity:
-                self.__dv = -self.delta_velocity
-            if self.velocity < self.min_velocity:
-                self.__dv = self.delta_velocity
-            self.velocity += self.__dv
+            elapsed_time = rospy.get_time() - start_time
+
+            if elapsed_time < 40:
```

Comment:
If I understand the rospy docs correctly, this means parking takes 40 seconds?

Reply:
Unfortunately the timer starts when the node is started. Since this node doesn't subscribe to the current velocity of the vehicle I added this workaround. 40 seconds results in about 10-15 seconds in the actual run (30 seconds are consumed by the loading of the map). This is probably not optimal as some computers might be faster/slower. Alternatively this node could subscribe to current velocity, wait for a change and then start a timer. Writing this I realize that using the velocity of the vehicle is probably the more robust approach.

Reply:
fixed in [d556b8e](https://github.com/ll7/paf22/pull/210/commits/d556b8ea0a8a784c0dcf05d11e1a526768c9dfce)